pandas/_testing/asserters.py

```diff
@@ -902,6 +902,7 @@ def assert_series_equal(
     >>> tm.assert_series_equal(a, b)
     """
     __tracebackhide__ = True
+    check_exact_index = False if check_exact is lib.no_default else check_exact
```

Comment:
Don't we want to default to check_exact=True for integers and false for floats?

Reply:
I wasn't a fan of doing the breaking change in a minor release, that's why I reverted back to 2.1.x behaviour for Index. We can introduce the more intelligent behavior for 3.0 if someone wants to do it

Reply:
Opened #57386 to track.